### ISSUE TITLE
JS driver - RqlDriverError: This shouldn't happen

### ISSUE THREAD

**Issue by neumino** [rethinkdb/rethinkdb#4240]
So if you fix #4239, you can end up with `RqlDriverError: This shouldn't happen`

This is a sneaky bug. You can reproduce it by opening a point change feed on an existing document, call `feed.next` and delete the document at the same time. As soon as `feed.next` is resolved, call `feed.close` 

What can happen is the following:

- Open the feed (get a partial response with the initial value)
- `feed.next` is resolved with the initial change
- Send the delete query
- Send the continue query for the changefeed
- Close the feed (`@_closeCb` is now defined)
- Get back the delete change with a `SUCCESS_PARTIAL`
- Reach error `This shouldn't happen`

Basically, having `_closeCb` defined doesn't mean that getting a `SUCCESS_PARTIAL` is impossible.
It fails about once out of 200 times for me on my desktop.

**Comment by danielmewes** [comment]
Thanks @neumino , that's an interesting bug.

@deontologician could you take a look at this please when you have some time?

**Comment by deontologician** [comment]
@neumino could you share your test script that's generating this problem?

**Comment by neumino** [comment]
@deontologician -- This test is one of those who were failing
https://github.com/neumino/reqlite/blob/master/test/changes.js#L185

**Comment by deontologician** [comment]
Thanks @neumino working on this now

**Comment by deontologician** [comment]
OK, I hacked it into a form that replicates the race condition often enough: https://gist.github.com/deontologician/24715e686c91ee7dbbdd

**Comment by deontologician** [comment]
This is up in CR 2914 by @AtnNn
